src/libraries/System.Data.OleDb/src/OleDb_Util.cs

```diff
@@ -687,7 +687,31 @@ internal static InvalidOperationException IDBInfoNotSupported()
         internal const string DbInfoKeywords = "DbInfoKeywords";
         internal const string Keyword = "Keyword";
 
-        internal static readonly bool IsRunningOnX86 = RuntimeInformation.ProcessArchitecture == Architecture.X86;
+        internal static readonly bool IsRunningOnX86;
```

Comment:
If initialization is done inline for the readonly fields, then IsRunningOnX86 initialization should move before its usage in the static initialization.

Reply:
Yes, I found that out the hard way.  That's why I ended up moving away from inlined initialization.  However, if that's the way it needs to be, then I am fine with that.

Reply:
It would look better to keep initialization inline